src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -117,7 +117,7 @@
         StringLexer("crly_bracket_open", "{", CodeSegment),
         StringLexer("crly_bracket_close", "}", CodeSegment),
         StringLexer("colon", ":", CodeSegment),
-        StringLexer("semicolon", ";", CodeSegment),
+        RegexLexer("semicolon", r";+", CodeSegment),
```

Comment:
This seems dangerous and don't understand why it should be necessary? Surely it should just lex as two semi-colons?

Also seems a bit brittle - what about dialects that override this or have non-semi-colon statement endings?

Reply:
Dangerous is probably a bit of an over statement. Just a little uncomfortable that we need to change the definition of `semicolon` to `semicolon(s)`. I'm not sure what assumptions are based on that to be honest.

To me this is not a lexing error - it's a parsing error. We should be able to lex this as multiple instances of semicolons. I don't get why it doesn't do this now and therefore why this change is necessary? Though I agree from your tests it does seem to work!

I just had a quick look and changing this in check_still_complete:

```py
    if initial_str != current_str:  # pragma: no cover
        raise RuntimeError(
            "Dropped elements in sequence matching! {!r} != {!r}".format(
                initial_str, current_str
            )
        )
```

to this (as well as importing `from sqlfluff.core.errors import SQLParseError` at the top):

```py
    if initial_str != current_str:
        raise SQLParseError(
            f"Could not parse: {current_str}",
             segment=unmatched_segments[0],
        )
```

Results in this:

```
% echo "SELECT 1 from table1;;" | sqlfluff parse -
...Failed to Parse...
==== parsing violations ====
L:   1 | P:  22 |  PRS | Could not parse: ;
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
```

Which seems a cleaner solution to me. And also handles other potential parsing issues.

Also the `# pragma: no cover` at the top shows this code is not expected to be encountered in any of our tests (but will now - thanks to your tests below) so think it is safe to change this code. All tests pass wit this alternative solution.

What do you think?

Reply:
Actually to be honest dropping that whole `check_still_complete` call completely returns the whole parse tree, which might be a better solution:

```py
% echo "SELECT 1 from table1;;" | sqlfluff parse -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    unparsable:                                               !! Expected: 'FileSegment'
[L:  1, P:  1]      |        raw:                                                  'SELECT'
[L:  1, P:  7]      |        whitespace:                                           ' '
[L:  1, P:  8]      |        raw:                                                  '1'
[L:  1, P:  9]      |        whitespace:                                           ' '
[L:  1, P: 10]      |        raw:                                                  'from'
[L:  1, P: 14]      |        whitespace:                                           ' '
[L:  1, P: 15]      |        raw:                                                  'table1'
[L:  1, P: 21]      |        raw:                                                  ';'
[L:  1, P: 22]      |        raw:                                                  ';'
[L:  1, P: 23]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'SELECT 1 from
                       | table1;;'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?
```

I wonder why this check is even here?!?! Especially without test cases? Time to dig through GitHub code to see when it was added and why...

Reply:
Seems it was added here, over two years ago: https://github.com/sqlfluff/sqlfluff/commit/6806c4b5e192a1af4f432d76cb91701b040650df by @alanmcruickshank 

To be honest I'd just be tempted to remove this function. I presume a lot has changed since then, it's not covered by a test case, and it's causing problems as this issue shows.

If it turns out this is here for a reason, then we can reinstate it at that point (with a test case!).

Reply:
But also happier with my earlier proposal, if wanna go more cautious route (maybe not a bad idea!).

Not sure of value in showing the full parse tree to be honest as not giving much info as it's not been able to be parsed, and earlier solution at least points the exact position (character 22) where as removing that function won't

Either way this shouldn't be seen much except on user error so don't think we need to worry about it too much as long as we solve the initial ask of not having the big, scary full stack trace.

So my vote is for my initial solution rather than removing this function completely.

Thoughts @adam-tokarski ? And @alanmcruickshank if free to comment?